File: week_1/day-3-two-pointer/variations_que/squared_sorted_array.js
const nums = [-7, -3, 2, 3, 11];

function sortedSquares(nums) {
    const n = nums.length;
    const res = new Array(n);
    let left = 0, right = n - 1;

    for (let i = n - 1; i >= 0; i--) {
        if (Math.abs(nums[left]) > Math.abs(nums[right])) {
            res[i] = nums[left] ** 2;
            left++;
        } else {
            res[i] = nums[right] ** 2;
            right--;
        }
    }
    return res;
}

console.log(sortedSquares(nums)); // [4, 9, 9, 49, 121]

File: week_1/day-3-two-pointer/variations_que/squared_sorted_array.py
nums = [-7, -3, 2, 3, 11]

def sortedSquares(nums):
    n = len(nums)
    res = [0] * n
    left, right = 0, n - 1

    for i in range(n - 1, -1, -1):
        if abs(nums[left]) > abs(nums[right]):
            res[i] = nums[left] ** 2
            left += 1
        else:
            res[i] = nums[right] ** 2
            right -= 1
    return res

print(sortedSquares(nums))  # Output: [4, 9, 9, 49, 121]
